connectors/camunda-message/src/main/java/io/camunda/connector/message/SendMessageConnectorFunction.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. Licensed under a proprietary license.
+ * See the License.txt file for more information. You may not use this file
+ * except in compliance with the proprietary license.
+ */
+package io.camunda.connector.message;
+
+import io.camunda.connector.api.annotation.OutboundConnector;
+import io.camunda.connector.api.outbound.OutboundConnectorContext;
+import io.camunda.connector.api.outbound.OutboundConnectorFunction;
+import io.camunda.connector.generator.dsl.BpmnType;
+import io.camunda.connector.generator.java.annotation.ElementTemplate;
+import io.camunda.connector.generator.java.annotation.ElementTemplate.ConnectorElementType;
+import io.camunda.connector.runtime.app.ZeebeClientContext;
+import io.camunda.zeebe.client.ZeebeClient;
+import io.camunda.zeebe.client.api.command.CorrelateMessageCommandStep1.CorrelateMessageCommandStep2;
+import io.camunda.zeebe.client.api.command.CorrelateMessageCommandStep1.CorrelateMessageCommandStep3;
+import io.camunda.zeebe.client.api.command.PublishMessageCommandStep1.PublishMessageCommandStep2;
+import io.camunda.zeebe.client.api.command.PublishMessageCommandStep1.PublishMessageCommandStep3;
+import io.camunda.zeebe.client.api.response.CorrelateMessageResponse;
+import io.camunda.zeebe.client.api.response.PublishMessageResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@OutboundConnector(
+    inputVariables = {
+      "messageName",
+      "correlationKey",
+      "variables",
+      "correlationType",
+      "tenantId",
+      "requestTimeout"
+    },
+    name = "message",
+    type = "io.camunda:sendMessage:1")
+@ElementTemplate(
+    id = "io.camunda.connectors.message.v1",
+    inputDataClass = SendMessageRequest.class,
+    name = "Send Message Connector",
+    icon = "send.svg",
+    elementTypes = {
+      @ConnectorElementType(
+          elementType = BpmnType.INTERMEDIATE_THROW_EVENT,
+          appliesTo = BpmnType.INTERMEDIATE_THROW_EVENT,
+          templateIdOverride = "io.camunda.connectors.message.intermediate.v1"),
+      @ConnectorElementType(
+          elementType = BpmnType.MESSAGE_END_EVENT,
+          appliesTo = BpmnType.MESSAGE_END_EVENT,
+          templateIdOverride = "io.camunda.connectors.message.end.v1"),
+      @ConnectorElementType(
+          elementType = BpmnType.SEND_TASK,
+          appliesTo = BpmnType.TASK,
+          templateIdOverride = "io.camunda.connectors.message.sendtask.v1")
+    })
+public class SendMessageConnectorFunction implements OutboundConnectorFunction {
+
+  private static final Logger LOG = LoggerFactory.getLogger(SendMessageConnectorFunction.class);
+
+  ZeebeClient zeebeClient;
```

Comment:
Please migrate to `CamundaClient`: the old zeebe client is flagged as deprecated and should not be used anymore